### PR TITLE
Neo4j

### LINKED ISSUE #921: Implement neo4j as a Go module

### PR DESCRIPTION
**Pull request by fbiville**
## What does this PR do?

This introduces support for Neo4j containers.

## Why is it important?

Neo4j is great database (I'm not biased at all /s) and deserves proper TestContainers support in Go.

## Related issues

- Closes #921

### CODE REVIEW

**Review by mdelapenya — COMMENTED** [review]
Wow, great PR @fbiville! Thank you very much for your time. I'm only missing docs on the public API

What do you think if we add docs in the same way that we do for localstack or pulsar? I'd start using any of them as example, describing the different options:
- https://golang.testcontainers.org/modules/localstack/#module-reference
- https://golang.testcontainers.org/modules/pulsar/#container-options

Once there, I think this is ready to go! 🚀

File: modules/neo4j/neo4j.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("neo4j://%s:%d", host, mappedPort.Int()), nil
+}
+
+// StartContainer creates an instance of the Neo4j container type
+func StartContainer(ctx context.Context, options ...Option) (*Neo4jContainer, error) {
+	settings := config{
+		imageCoordinates: fmt.Sprintf("%s:%s", defaultImageName, defaultTag),
```
Comment (mdelapenya) [inline modules/neo4j/neo4j.go:43]: Could we prepend "docker.io" as registry in a explicit manner? That will be inclusive with other container runtimes, not only for Docker.

**Comment by fbiville** [comment]
Apologies, I have not been able to find time to work on this issue again this week.
I'll be in vacation next week. I'll do my best to update this as soon as I'm back.

File: modules/neo4j/neo4j.go (hunk 1 above)
Comment (fbiville) [reply modules/neo4j/neo4j.go]: Fixed in 2457831302d3466bd6b64041131b043dd95a770e

**Comment by fbiville** [comment]
I started with a minimal documentation page.
There is also a list of other features to support listed in the initial comment of this PR.

Shall I complete the docs for the current feature set and work on the other features in subsequent PRs?
(I'll be on vacation next week, but I can resume the work when I'm back)

**Comment by mdelapenya** [comment]
> Shall I complete the docs for the current feature set and work on the other features in subsequent PRs?

As the neo4j module won't appear until the next release, I think it's fine to set this as ready-for-review and eventually merge this PR, and continue with the extensions in follow-up PRs.

Thanks for contributing this module! Looks promising!

**Comment by mdelapenya** [comment]
I did not mention, but we are writing down a design doc for modules so, once approved, we will eventually update all existing modules to that design.

This wants to reinforce the message of "progress, simple perfection" that I learnt at my time at Elastic :) So go for it adding incremental changes in small batches

**Review by mdelapenya — APPROVED** [review]
LGTM! Thanks for the module for such a great technology!

As I said, we are planning to standardise how to create options, so we probably update the code in a follow-up PR with some design changes, but we prefer seeing progress here merging the PR and iterating in small chunks later. Will ping you in there.

Thanks again for your work in this PR

**Comment by fbiville** [comment]
@mdelapenya now rebased

**Review by mdelapenya — APPROVED** [review]

**Pull request merged**